test/unit/node/settings.test.ts

```diff
@@ -0,0 +1,65 @@
+import { logger } from "@coder/logger"
+import { promises as fs } from "fs"
+import { SettingsProvider, CoderSettings } from "../../../src/node/settings"
+import { clean, mockLogger, tmpdir } from "../../utils/helpers"
+
+describe("settings", () => {
+  describe("with invalid JSON in settings file", () => {
+    const testName = "settingsInvalidJSON"
+    let testDir = ""
+    let mockSettingsFile = "coder.json"
+    let pathToMockSettingsFile = ""
+
+    beforeEach(async () => {
+      mockLogger()
+      await clean(testName)
+      testDir = await tmpdir(testName)
+      pathToMockSettingsFile = `${testDir}/${mockSettingsFile}`
+      // Missing a quote, which makes it invalid intentionally
+      await fs.writeFile(pathToMockSettingsFile, '{"fakeKey":true,"helloWorld:"test"}')
+    })
+    afterEach(async () => {
+      await fs.rm(testDir, { recursive: true, force: true })
+      jest.clearAllMocks()
+    })
+    it("should log a warning", async () => {
+      const settings = new SettingsProvider<CoderSettings>(pathToMockSettingsFile)
+      await settings.read()
+      // This happens when we can't parse a JSON (usually error in file)
+      expect(logger.warn).toHaveBeenCalledWith("Unexpected token t in JSON at position 29")
+    })
+  })
+  describe("with invalid settings file path", () => {
+    const testName = "settingsInvalidPath"
+    let testDir = ""
+    let mockSettingsFile = "nonExistent.json"
+    let pathToMockSettingsFile = ""
+
+    beforeEach(async () => {
+      mockLogger()
+      await clean(testName)
+      testDir = await tmpdir(testName)
+      // Add hello so it's a directory that doesn't exist
+      // NOTE: if we don't have that, it fails the test
+      // That's because it will write a file if it doesn't exist
+      // but it throws if there's a directory in the path that
+      // doesn't exist.
+      pathToMockSettingsFile = `${testDir}/hello/${mockSettingsFile}`
```

Comment:
Recommend using `path.join` for cross-platform compat.

Reply:
Ahh good call!